File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> romanValues{
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };
        int result = romanValues[s.back()];
        for (int i = 0; i < s.size() - 1; ++i) {
            int sign = romanValues[s[i]] < romanValues[s[i + 1]] ? -1 : 1;
            result += sign * romanValues[s[i]];
        }
        return result;
    }
};

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_values = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        integer_value = 0
        prv = 0
        for i in range(len(s)):
            cur = roman_values[s[i]]
            if cur > prv and i > 0:
                integer_value += cur - 2 * prv
            else:
                integer_value += cur
            prv = cur
        return integer_value
